docs/bookstore/page-1/send-review-comment-to-broker.md

```diff
@@ -0,0 +1,423 @@
+
+![Image](images/image9.png)
+
+A Trigger is able to forward the event to the correct destination based on the [CloudEvent's attributes](https://knative.dev/docs/eventing/#:~:text=Knative%20Eventing%20uses%20standard%20HTTP%20POST%20requests%20to%20send%20and%20receive%20events%20between%20event%20producers%20and%20sinks.%20These%20events%20conform%20to%20the%20CloudEvents%20specifications%2C%20which%20enables%20creating%2C%20parsing%2C%20sending%2C%20and%20receiving%20events%20in%20any%20programming%20language.){:target="_blank"}. It is the connector between the Broker and the event destination.
+
+A Filter in the Trigger will **filter the events based on the filter condition**. You will specify your filter condition in the Trigger’s YAML file. **If no filter is specified, the Trigger will forward all the events that the Broker received.**
+
+![Image](images/image18.png)
+
+There is also a concept called [Channel](https://knative.dev/docs/eventing/channels/){:target="_blank"} in Knative, and generally speaking, you can treat Broker & Trigger without filter the same as Channel & Subscription.
+
+Learn more about Broker & Trigger [here](https://knative.dev/docs/eventing/brokers/){:target="_blank"}!
+
+**Create a Trigger:**
+
+![Image](images/image20.png)
+
+Here we are creating a Trigger that will send all the events to event-display.
+
+![Image](images/image4.png)
+
+- 1: Create a new YAML file named `200-log-trigger.yaml` and add the following content:
+
+???+ abstract "_node-server/config/200-log-trigger.yaml_"
+    ```yaml 
+    # This Trigger subscribes to the Broker and will forward all the events that it received to event-display.
+    apiVersion: eventing.knative.dev/v1
+    kind: Trigger
+    metadata:
+      name: log-trigger
+    spec:
+      broker: bookstore-broker
+      subscriber:
+        ref:
+          apiVersion: v1
+          kind: Service
+          name: event-display
+    ```
+
+- 2: Apply the YAML file:
+
+```bash
+kubectl apply -f node-server/config/200-log-trigger.yaml
+```
+
+You will see the following output:
+```
+trigger.eventing.knative.dev/log-trigger created
+```
+
+???+ success "Verify"
+    Running the following command to list the Triggers:
+    ```bash
+    kubectl get triggers
+    ```
+    The Trigger `log-trigger` should have `READY` status as `True`.
+
+    
+    ```
+    NAME                BROKER             SUBSCRIBER_URI                                                       AGE     READY   REASON
+    log-trigger         bookstore-broker   http://event-display.default.svc.cluster.local                       6m2s    True    
+    ```
+
+
+### **Validate**
+
+![Image](images/image8.png)
+
+Open the logs of the event-display with the following command:
+
+```bash
+kubectl logs event-display-XXXXX -f
+```
+
+???+ success "Verify"
+    Type something in the comment box in the UI and click the submit button. The comment should appear in the event-display service with the following output:
```

Comment:
When I was doing the verification, trying to send events through Broker and verifying it on the event display side the UI got stuck on "The CloudEvent has been dispatched by Broker, waiting for an acknowledgement." and I've got this error in the port-forward command `kubectl port-forward svc/node-server-svc 8080:80`

```
E0620 15:19:27.460988  906586 portforward.go:409] an error occurred forwarding 8080 -> 8000: error forwarding port 8000 to pod cc2bcc501cda7aef222ed364b038f7c7152003d011351aa781889e56a60d3bbf, uid : failed to find sandbox "cc2bcc501cda7aef222ed364b038f7c7152003d011351aa781889e56a60d3bbf" in store: not found
error: lost connection to pod
```

Reply:
After re-running `kubectl port-forward svc/node-server-svc 8080:80` and re-sending the comment, I get the event in the event display and the UI is not stuck and finished

Reply:
Is that because when you first time running the port-forwarding command, the pods aren't becoming ready yet? If that's the case, I can add a warn box/Troubleshooting box there saying if this error happens, simply re-run the portforwarding command.

Reply:
this worked here all directly